app/lib/features/home/providers/navigation.dart

```diff
@@ -41,23 +42,51 @@ final spaceItemsProvider = FutureProvider.autoDispose
         ),
         location: '/$roomId',
       ),
-      data: (info) => SidebarNavigationItem(
-        icon: SpaceParentBadge(
-          roomId: roomId,
-          child: ActerAvatar(
-            uniqueId: roomId,
-            displayName: info.displayName,
+      data: (info) => canonicalParent.maybeWhen(
+        data: (parent) => SidebarNavigationItem(
+          icon: ActerAvatar(
             mode: DisplayMode.Space,
-            avatar: info.getAvatarImage(),
+            avatarInfo: AvatarInfo(
+              uniqueId: roomId,
+              displayName: info.displayName,
+              avatar: info.getAvatarImage(),
+            ),
+            avatarsInfo: parent != null
+                ? [
+                    AvatarInfo(
+                      uniqueId: parent.space.getRoomIdStr(),
+                      displayName: parent.profile.displayName,
+                      avatar: parent.profile.getAvatarImage(),
+                    ),
+                  ]
+                : [],
```

Comment:
other than this, the code is the same, no? And we don't care about any difference between loading and error... so why not go for `canonicalParent.valueOrNull != null ...` allowing us to collapse that entire code and remove the `orElse` ...